src/benchmarklib/benchmark_config.hpp

```diff
@@ -13,21 +13,31 @@ namespace opossum {
  */
 enum class BenchmarkMode { Ordered, Shuffled };
 
+/**
+ * Clustering to apply after loading the benchmark data:
+ *   - "None" does not apply any clustering. For most TPC-H and TPC-DS that means that the order of the data generator
+ *     is used.
+ *   - "TPCHPruning" is a clustering that improves the pruning rates in TPC-H.
+ */
+enum class ClusteringConfiguration { None, TPCHPruning };
+
```

Comment:
Seeing the whole picture here, I believe that this should not be in the benchmark config. After all, we do not store the scale factor in here, either.

Reply:
I agree. First it was part of the TPCH benchmark only. But that leads to add additional parameter for the data generator and some ugly passing to the data loading. And I think the check during the binary loading is quite a help.
Any ideas how to ease the passing around of the config without adding to the benchmark config?

Reply:
> But that leads to add additional parameter for the data generator and some ugly passing to the data loading.

Any worse than what we are currently doing with the scale factor?

> And I think the check during the binary loading is quite a help.

But the binary loading is in the TPCHTableGenerator, too. Do you mean the final sanity check? Might a bool "expecting_some_clustering" be sufficient?

Reply:
I think it's ok as it is now. Instead of passing the scale factor, it's now the scale factor and the clustering.